java-checks/src/test/files/checks/AssertionFailInCatchBlockCheck.java

```diff
@@ -0,0 +1,84 @@
+import org.junit.Assert;
+import org.junit.Test;
+
+public class MyTest {
+
+  static {
+    try {
+      File.createTempFile(null, null);
+    } catch (IOException e) {
+      Assert.fail(e.getMessage()); // Compliant, corner case accepted ... and MethodMatcher doesn't pass through this case
```

Comment:
This is an error of semantic : write java.io.IOException and the case will be raised.